### PR TITLE
Capture logs from of-builder

### PR DESCRIPTION
**Pull request by alexellis**
## Description

A struct is now returned as JSON from of-builder containing an
array of lines from buildkit and the imagename/status.

Tested locally with Docker for Mac - I received logs on a
positive/negative build.

Signed-off-by: Alex Ellis (VMware) <alexellis2@gmail.com>

## Checklist:

I have:

- [x] updated the documentation if required
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: sdk/build.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package sdk
+
+type Build struct {
```
Comment (kvuchkov) [inline sdk/build.go]: Isn't BuildResult a more suitable struct for the sdk, as it is returned by the build service?

Comment (alexellis) [reply sdk/build.go]: Yes for sure

File: of-builder/Dockerfile [hunk 2]
```diff
@@ -8,18 +8,17 @@ RUN go build -o /usr/bin/of-builder .
 
 FROM alpine:3.7
 
-RUN addgroup -S app && adduser -S -g app app
-RUN mkdir -p /home/app
+RUN addgroup -S app && adduser app -S -G app \
```
Comment (kvuchkov) [inline of-builder/Dockerfile]: Adding the user to a group is causing an "operation not permitted" when untar tries to lchown in the tmp folder, e.g. "/tmp/buildctx584615953/context".

File: of-builder/main.go [hunk 3]
```diff
@@ -22,12 +22,23 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+var (
+	lchownEnabled bool
```
Comment (kvuchkov) [inline of-builder/main.go]: If we revert the group of the user, should we remove the flag?

File: buildshiprun/handler.go [hunk 4]
```diff
@@ -365,3 +378,11 @@ func getRegistryAuthSecret() string {
 	}
 	return ""
 }
+
+// BuildResult represents a successful Docker build and
+// push operation to a remote registry
+type BuildResult struct {
```
Comment (kvuchkov) [inline buildshiprun/handler.go]: Do you think we need to extract this struct in sdk?

Comment (alexellis) [reply buildshiprun/handler.go]: Sure

File: of-builder/main.go (hunk 3 above)
Comment (alexellis) [reply of-builder/main.go]: We may need it again for testing, I'll leave for now but unused.

**Comment by alexellis** [comment]
Ugh

**Comment by alexellis** [comment]
Messed this PR branch up and merged the same via #102

**Pull request closed without merging**